pkg/bootstrap/api/types.go

```diff
@@ -80,6 +86,12 @@ const (
 	// authenticate as. The full username given is "system:bootstrap:<token-id>".
 	BootstrapUserPrefix = "system:bootstrap:"
 
-	// BootstrapGroup is the group bootstrapping bearer tokens authenticate in.
-	BootstrapGroup = "system:bootstrappers"
+	// BootstrapGroupPattern is the valid regex pattern that all groups
+	// assigned to a bootstrap token by BootstrapTokenExtraGroupsKey must match.
+	// See also ValidateBootstrapGroupName().
+	BootstrapGroupPattern = "system:bootstrappers:[a-zA-Z0-9:-]{0,255}[a-zA-Z0-9]"
```

Comment:
I can make this more restrictive if we want. These groups will probably primarily come from things we generate inside kubeadm, where we follow the `kebab-case` convention used by all the other group names.

 @luxas does that work for you?

Maybe:
`"system:bootstrappers:[a-z0-9:-]{0,255}[a-z0-9]"` (no uppercase)

Or even:
`"system:bootstrappers:[a-z:-]{0,255}[a-z]"` (no numbers either)